source/elements/oneDAL/source/algorithms/decomposition/pca.rst

```diff
@@ -1,3 +1,447 @@
+
+Given the training set :math:`X = \{ x_1, \ldots, x_n \}` of
+:math:`p`-dimensional samples and the number of principal components :math:`r`,
+the problem is to compute :math:`r` principal directions (:math:`p`-dimensional
+eigenvectors) for the training set. The eigenvectors can be grouped into the
+:math:`r \times p` matrix :math:`T` that contains one eigenvectors in each row.
+
+-----------------------
+Covariance-based method
+-----------------------
+[TBD]
+
+----------------
+SVD-based method
+----------------
+[TBD]
+
+-------------------
+Sign-flip technique
+-------------------
+Eigenvectors computed by some eigenvalue solvers are not uniquely defined due to
+sign ambiguity. To get the deterministic result, a sign-flip technique should be
+applied. One of the sign-flip techniques proposed in [Bro07]_ requires the
+following modification of eigenvectors matrix :math:`T`,
+
+.. math::
+   \hat{T}_i = T_i \cdot \mathrm{sgn}(\max_{1 \leq j \leq p } |{T}_{ij}|), \quad 1 \leq i \leq r,
+
+where :math:`T` is eigenvectors matrix, :math:`T_i` is :math:`i`-th row,
+:math:`T_{ij}` is the element in the :math:`i`-th and :math:`j`-th column,
+:math:`\mathrm{sgn}(\cdot)` is signum function,
+
+.. math::
+   \mathrm{sgn}(x) =
+      \begin{cases}
+         -1, & x < 0, \\
+          0, & x = 0, \\
+          1, & x > 0.
+      \end{cases}
+
+.. note::
+   The sign-flip technique described above is provided as an example. oneDAL
+   spec does not require implementation of this sign-flip technique.
+   Implementers can choose an arbitrary technique that modifies sign of the
+   eigenvectors or use deterministic algorithm for eigenvectors computations.
+
+
+-------------
+Usage example
+-------------
+::
+
+   onedal::pca::model run_training(const onedal::table& data) {
+
+      const auto pca_desc = onedal::pca::desc<float>{}
+         .set_component_count(5)
+         .set_is_deterministic(true);
+
+      const auto result = onedal::train(pca_desc, data);
+
+      print_table("means", result.get_means());
+      print_table("variances", result.get_variances());
+      print_table("eigenvalues", result.get_eigenvalues());
+      print_table("eigenvectors", result.get_model().get_eigenvectors());
+
+      return result.get_model();
+   }
+
+
+::
+
+   onedal::table run_inference(const onedal::pca::model& model,
+                               const onedal::table& new_data) {
+
+      const auto pca_desc = onedal::pca::desc<float>{}
+         .set_component_count(model.get_component_count());
+
+      const auto result = onedal::infer(pca_desc, model, new_data);
+
+      print_table("labels", result.get_transformed_data());
+   }
+
+
+---
+API
+---
+Methods
+-------
+::
+
+   namespace method {
+      struct cov {};
+      struct svd {};
+      using by_default = svd;
+   } // namespace method
+
+
+.. namespace:: onedal::pca::method
+
+.. struct:: cov
+
+   Tag-type that denotes `Covariance-based method`_.
```

Comment:
To all similar descriptions: makes sense to remove a full stop at the end as it is not a full sentence